### PR TITLE
Middleware for Workspace member Check

### PR DESCRIPTION
**Pull request by rnavaneeth992**
Issue: #44 

Added isWorkspaceMember.middleware.js

### CODE REVIEW

File: src/middlewares/isWorkspaceMember.middleware.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+exports.ensureUserIsMember = async (req, res, next) => {
+    try {
+        const user = req.body.user;
+        const workspace = await Workspace.findById(req.params.id);
+
+        if (!workspace) {
+            return response_404(res, "Workspace doesn't exist");
+        }
+
+        const isMember = await Channel.exists({
```
Comment (automatefreely) [inline src/middlewares/isWorkspaceMember.middleware.js]: You can get the channel directly from workspace. That will be less time complex i think.

**Review by automatefreely — COMMENTED** [review]
Your method is good but querying all the channels is not efficient

File: src/middlewares/isWorkspaceMember.middleware.js (hunk 1 above)
Comment (rnavaneeth992) [reply src/middlewares/isWorkspaceMember.middleware.js]: Ok Got it, I will do it

Comment (rnavaneeth992) [reply src/middlewares/isWorkspaceMember.middleware.js]: @automatefreely Can you check now?

**Comment by rnavaneeth992** [comment]
@automatefreely Please Merge this
Then only you can assign me for another task

I requested first for that

File: src/middlewares/isWorkspaceMember.middleware.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+exports.ensureUserIsMember = async (req, res, next) => {
+    try {
+        const user = req.body.user;
+        const workspace = await Workspace.findById(req.params.id);
+
+        if (!workspace) {
+            return response_404(res, "Workspace doesn't exist");
+        }
+
+        const isMember = workspace.channels.some(channel => channel.members.includes(user._id));
```
Comment (automatefreely) [inline src/middlewares/isWorkspaceMember.middleware.js]: You can not access channel.member

**Review by automatefreely — COMMENTED** [review]
Also admin of the workspace is a user

File: src/middlewares/isWorkspaceMember.middleware.js (hunk 2 above)
Comment (rnavaneeth992) [reply src/middlewares/isWorkspaceMember.middleware.js]: Also admin of the workspace is a user 

You can not access channel.member

If the user is not member, send a response with status code 403 and message "You are not authorized to perform this action"A user is consider a member of a workspace if he is member of at least a channel belonging to it. Admin is a member also.

bro in this case I have to check members only right?
or I have to check createdBy(channels) is request param

Comment (rnavaneeth992) [reply src/middlewares/isWorkspaceMember.middleware.js]: Can You explain this part alone? please @automatefreely

File: src/middlewares/isWorkspaceMember.middleware.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const Channel = require("../models/channel.models");
+const Workspace = require("../models/workspace.models");
+const {
+    response_403,
+    response_500,
+    response_404,
+} = require("../utils/responseCodes.utils");
+
+exports.ensureUserIsMember = async (req, res, next) => {
```
Comment (automatefreely) [inline src/middlewares/isWorkspaceMember.middleware.js]: make the name isWorksapceMember

**Review by automatefreely — COMMENTED** [review]
Do the small change

**Pull request merged**